app/core/Engine/controllers/snaps/snap-controller-init.ts

```diff
@@ -0,0 +1,124 @@
+import { SnapController } from '@metamask/snaps-controllers';
+import { hasProperty } from '@metamask/utils';
+import { ControllerInitFunction } from '../../types';
+import {
+  SnapControllerInitMessenger,
+  SnapControllerMessenger,
+} from '../../messengers/snaps';
+import {
+  EndowmentPermissions,
+  ExcludedSnapEndowments,
+  ExcludedSnapPermissions,
+  detectSnapLocation,
+} from '../../../Snaps';
+import {
+  Encryptor,
+  LEGACY_DERIVATION_OPTIONS,
+  pbkdf2,
+} from '../../../Encryptor';
+import { KeyringTypes } from '@metamask/keyring-controller';
+import { selectBasicFunctionalityEnabled } from '../../../../selectors/settings';
+import { store } from '../../../../store';
+import PREINSTALLED_SNAPS from '../../../../lib/snaps/preinstalled-snaps';
+
+/**
+ * Initialize the Snap controller.
+ *
+ * @param request - The request object.
+ * @param request.initMessenger - The init messenger. This has access to
+ * different functions than the controller messenger, and should be used for
+ * initialization purposes only.
+ * @param request.controllerMessenger - The messenger to use for the controller.
+ * @param request.persistedState - The persisted state of the extension.
+ * @returns The initialized controller.
+ */
+export const snapControllerInit: ControllerInitFunction<
+  SnapController,
+  SnapControllerMessenger,
+  SnapControllerInitMessenger
+> = ({ initMessenger, controllerMessenger, persistedState }) => {
+  const requireAllowlist = process.env.METAMASK_BUILD_TYPE === 'main';
+  const disableSnapInstallation = process.env.METAMASK_BUILD_TYPE === 'main';
```

Comment:
```suggestion
  const disableSnapInstallation = process.env.METAMASK_BUILD_TYPE !== 'flask';
```
There's an incoming beta build type and we've changed the condition in that branch